### ISSUE TITLE
As a contributor, I want Composer to run under Node.js on Hyperledger Fabric v1.1

### ISSUE THREAD

**Issue by dselman** [hyperledger/composer#61]
This describes removing our current javascript interpreter (duktape) and being able to exploit the capabilities of Hyperledger Fabric 1.1 where native node.js chaincode is supported. This feature provides no new features over what can currently be done so will not include support for TP functions to use 3rd party npm modules (this is a future item).

## Tasks and Status
(Now currently awaiting official 1.1 drops from HLF)
- [x] Implement Bridge code prototype
- [x] System tests to handle chicken/egg situation regarding testing code
- [x] Complete Unit tests for 100% code coverage
- [x] (Optional) migrate Bridge code to node 8 async/await syntax
- [x] **Dependency** dependent on #2100 (upgrade to node v8) being done in conjunction will be done with the release of BN cards.
- [x] require official HLF images in order to progress
- [x] Merge branch to master, fixing conflicts (hlfconnection.js, package.json)
- [x] Move Dev Servers up to HLF published drops of 1.1

**Comment by sstone1** [comment]
This work is being done in the Hyperledger Fabric project under [FAB-2331](https://jira.hyperledger.org/browse/FAB-2331).

**Comment by davidkel** [comment]
Status update for interested parties
- this is dependent on Hyperledger Fabric v1.1 which will ship with node.js chaincode support. Current discussions on the fabric-release/fabric-maintainers channel say they are targetting a November 15th release date.
- HLF chaincode support is going well with an early adopters repo available to try
- Composer is able to run in this environment and apart from the query system tests relating to limit and skip which are failing it runs well.

**Comment by davidkel** [comment]
Current status update
- composer implementation is complete. A few more tests required and a quick revisit on error handling and logging required
- automated build is complete but slightly unreliable due to using a npm proxy server which may have some issues, may need to find a more reliable alternative but there may not be one
- We will want to move both end user and composer development environments to Node 8 to be consistent with chaincode development
- We then will need to obtain official alpha images from HLF and then move both the build system and composer-dev-servers to this official alpha release. At the same time we will drop support for the old duktape implementation. This will not be a migration story. It will be a drop one environment and present a completely new environment for which people will probably need to start from scratch.

**Comment by davidkel** [comment]
Runtime support has now been merged into master.
